sd_dynamic_prompts/utils.py

```diff
@@ -0,0 +1,22 @@
+import logging
+
+
+def get_logger(name):
+    from modules.shared import opts
+
+    is_debug = getattr(opts, "is_debug", False)
+    logger = logging.getLogger(name)
+
+    if is_debug:
+        logger.setLevel(logging.DEBUG)
+    else:
+        logger.setLevel(logging.INFO)
+
+    sh = logging.StreamHandler()
+    formatter = logging.Formatter(
+        "%(asctime)s - %(name)s - %(levelname)s - %(message)s",
+    )
+    sh.setFormatter(formatter)
+    logger.addHandler(sh)
+
+    return logger
```

Comment:
IMO this is misguided – a logger shouldn't configure its own handler and formatter, the application hosting the library using a logger should. (What if I wanted my log messages to use JSON formatting?)

Reply:
agreed, except that auto1111 doesn't handle logging in any way that is useful. Right now it's a free-for-all